engine/runtime/src/main/java/org/enso/interpreter/node/callable/CaptureCallerInfoNode.java

```diff
@@ -55,7 +55,7 @@ public LocalScope getLocalScope() {
       return localScope;
     }
 
-    public ModuleScope getModuleScope() {
+    public ModuleScope.Builder getModuleScope() {
```

Comment:
Why does `ScopeInfo` needs a `Builder`? Shouldn't all the scope already be built before execution even gets here?

Reply:
Yes, I know this one is not obvious but this one is needed because of `EvalNode`'s `parseExpression` which builds `ClosureRootNode`, which in turn requires the builder because that's the only thing that `IrToTruffle` has when it creates those as well. catch-22.